pymc_marketing/mmm/base.py

```diff
@@ -404,6 +420,97 @@ def plot_posterior_predictive(
+        ----------
+        original_scale : bool, optional
+            Whether to plot in the original scale.
+        ax : plt.Axes, optional
+            Matplotlib axis object.
+
+        Returns
+        -------
+        plt.Figure
+        """
+        posterior_predictive_data: Dataset = self.posterior_predictive
+
+        target = np.asarray(
+            transform_1d_array(self.get_target_transformer().transform, self.y)
+        )
+
+        if len(target) != len(posterior_predictive_data.date):
+            raise ValueError(
+                "The length of the target variable doesn't match the length of the date column. "
+                "If you are computing out-of-sample errors, please overwrite `self.y` with the "
+                "corresponding (non-transformed) target variable."
+            )
+
+        target_broadcast = np.atleast_1d(target)[np.newaxis, np.newaxis, ...]
+        errors = target_broadcast - posterior_predictive_data
+
+        errors_hdi_94: DataArray = az.hdi(ary=errors, hdi_prob=0.94)[self.output_var]
+        errors_hdi_50: DataArray = az.hdi(ary=errors, hdi_prob=0.50)[self.output_var]
+
+        if original_scale:
+            errors = apply_sklearn_transformer_across_dim(
+                data=errors,
+                func=self.get_target_transformer().inverse_transform,
+                dim_name="date",
+            )
+
+            errors_hdi_94 = self.get_target_transformer().inverse_transform(
+                Xt=errors_hdi_94
+            )
+            errors_hdi_50 = self.get_target_transformer().inverse_transform(
+                Xt=errors_hdi_50
+            )
+
+        if ax is None:
+            fig, ax = plt.subplots(**plt_kwargs)
+        else:
+            fig = ax.figure
+
+        if self.X is not None and self.y is not None:
+            ax.fill_between(
+                x=posterior_predictive_data.date,
+                y1=errors_hdi_94[:, 0],
+                y2=errors_hdi_94[:, 1],
+                color="C3",
+                alpha=0.2,
+                label="$94\%$ HDI",  # noqa: W605
+            )
+
+            ax.fill_between(
+                x=posterior_predictive_data.date,
+                y1=errors_hdi_50[:, 0],
+                y2=errors_hdi_50[:, 1],
+                color="C3",
+                alpha=0.3,
+                label="$50\%$ HDI",  # noqa: W605
+            )
+
+            ax.plot(
+                posterior_predictive_data.date,
+                errors[self.output_var].mean(dim=("chain", "draw")).to_numpy(),
+                color="C3",
+                label="Errors Mean",
+            )
+
+            ax.axhline(y=0.0, linestyle="--", color="black", label="zero")
+            ax.legend()
+            ax.set(
+                title="Errors Posterior Distribution",
+                xlabel="date",
+                ylabel="true - predictions",
+            )
+        else:
+            raise RuntimeError("The model hasn't been fit yet, call .fit() first")
```

Comment:
Would the posterior_predictive exist if X and y don't exist? 
Just think we could raise this earlier. Thoughts?

Reply:
Make sense! Changed!